From 1fde1dd92030e3af1ec14e5a63b37f71ae73ced5 Mon Sep 17 00:00:00 2001
From: Fei Yang <fei.yang@intel.com>
Date: Fri, 12 Sep 2014 15:08:12 -0700
Subject: [PATCH] dwc3:hack stall otg_srp/hnp_reqd test mode

From OTG2.0 Spec, 6.4.3.1.1 Feature Bit 'otg_srp_reqd'
"A B-device not capable of SRP shall stall the SetFeature
request." 6.5.3.1.2 Feature Bit 'otg_hnp_reqd' "A B-device
not capable of HNP, shall STALL the Set Feature request"

We don't support SRP/HNP, so stall related setfeature request.

Change-Id: I4c2d2d585ddbb28ca2ed1313373513777186bf6b
Signed-off-by: Wu, Hao <hao.wu@intel.com>
(cherry picked from commit d9919db23c91a820348a480ebd0f29bab456b30f)

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/usb/dwc3/ep0.c |   15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/dwc3/ep0.c b/drivers/usb/dwc3/ep0.c
index ef79da6..cf7399c 100644
--- a/drivers/usb/dwc3/ep0.c
+++ b/drivers/usb/dwc3/ep0.c
@@ -430,8 +430,19 @@ static int dwc3_ep0_handle_feature(struct dwc3 *dwc,
 			if (!set)
 				return -EINVAL;
 
-			dwc->test_mode_nr = wIndex >> 8;
-			dwc->test_mode = true;
+			/* Stall SRP/HNP test modes */
+			switch (wIndex >> 8) {
+			case TEST_J:
+			case TEST_K:
+			case TEST_SE0_NAK:
+			case TEST_PACKET:
+			case TEST_FORCE_EN:
+				dwc->test_mode_nr = wIndex >> 8;
+				dwc->test_mode = true;
+				break;
+			default:
+				return -EINVAL;
+			}
 			break;
 		default:
 			return -EINVAL;
-- 
1.7.9.5

